### PR TITLE
init

### PR DESCRIPTION
**Pull request by jabajke**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:slim
+
+ENV PYTHONBUFFERED=1
+
+WORKDIR usr/src/app
+
```
Comment (shprexen) [inline Dockerfile:6]: add  COPY command for requirements.txt as a separate command. So when you'll add new dependencies in requirements.txt docker image will be rebuilt

File: autohouse/models.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class Autohouse(models.Model):
+    title = models.CharField(max_length=255)
+    location = CountryField()
+    prefer_characteristic = models.ManyToManyField('Characteristic')
+    is_active = models.BooleanField(default=True)
+
+
+class Auto(models.Model):
```
Comment (shprexen) [inline autohouse/models.py:18]: Let's name that model as Car

File: autohouse/models.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+from .validators import CharacteristicJSONValidationSchema
+from .schemas import EnumSchemas
+
+
+class Autohouse(models.Model):
+    title = models.CharField(max_length=255)
+    location = CountryField()
+    prefer_characteristic = models.ManyToManyField('Characteristic')
+    is_active = models.BooleanField(default=True)
```
Comment (shprexen) [inline autohouse/models.py:15]: create abstract model with fields, such as is_active, created_at and updated_at and inherit from it

**Review by shprexen — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (jabajke) [reply Dockerfile:6]: done

File: autohouse/models.py (hunk 2 above)
Comment (jabajke) [reply autohouse/models.py]: done

File: autohouse/models.py (hunk 3 above)
Comment (jabajke) [reply autohouse/models.py]: done

File: autohouse/models.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        abstract = True
+
+
+class Autohouse(CommonInfo):
+    title = models.CharField(max_length=255)
+    location = CountryField()
+    prefer_characteristic = models.ManyToManyField('Characteristic')
+
+class Car(CommonInfo):
+    characteristic = models.ForeignKey('Characteristic', on_delete=models.CASCADE)
```
Comment (shprexen) [inline autohouse/models.py:25]: in the future will it be convenient to add filters on endpoint using such field?
I think it's better to store characteristics in `Car`, i.e. VIN, model, manufacturer, horse power and etc.
But you can use `JSONField` in `Autohouse` model to represent preferred characteristics

Comment (jabajke) [reply autohouse/models.py]: done

File: autohouse/models.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+class Car(CommonInfo):
+    brand = models.CharField(max_length=50)
+    model = models.CharField(max_length=50)
+    horse_power = models.PositiveSmallIntegerField()
+    color = models.CharField(max_length=50)
+    year_of_issue = models.PositiveSmallIntegerField(validators=[
+        MinValueValidator(1800), MaxValueValidator(datetime.now().year)
+    ], help_text="Use <YYYY> as date format")
+    transmission_type = models.CharField(max_length=50)
```
Comment (shprexen) [inline autohouse/models.py:40]: Let's use here django choices, create choices with all types of transmission

**Review by shprexen — CHANGES_REQUESTED** [review]

File: autohouse/models.py (hunk 5 above)
Comment (jabajke) [reply autohouse/models.py]: done

**Review by shprexen — APPROVED** [review]

**Pull request merged**